### ISSUE TITLE
 startZap when invoked stales and never goes past the "starting stage"

### ISSUE THREAD

**Issue by urand0m** [jenkinsci/zap-pipeline-plugin#4]
Hello,

I have been trying to run as explained in the example of the wiki page, once started it just stale and never goes past the running stage in a declarative pipeline.
Maybe it's me but could you help me with this?
Thanks for your help

```
[owaspZapTest] $ /usr/share/zaproxy/zap.sh -daemon -host 0.0.0.0 -port 9091 -config api.disablekey=true -config api.addrs.addr.regex=true -config api.addrs.addr.name=10.* -config connection.timeoutInSecs=600
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (scan)
Stage "scan" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] script
[Pipeline] {
[Pipeline] archiveZap
zap: Archiving results...
zap: Failed to archive results
zap: Failed to shutdown ZAP (it's not running?)
```

**Comment by vrondakis** [comment]
Hey
It may be useful to check your Jenkins log file. Mine is in /var/log/jenkins/jenkins.log.
Run a build & there should be an error

**Comment by urand0m** [comment]
Hi thanks. 
so I am checking the jenkins logs : 

`zap: Started successfully
zap: Checking ZAP is alive at 127.0.0.1:8088
zap: Attempting to connect to ZAP at 127.0.0.1:8088
zap: Attempting to connect to ZAP at 127.0.0.1:8088
zap: Attempting to connect to ZAP at 127.0.0.1:8088
zap: Failed to start ZAP on port 8088
zap: Archiving results...
com.mashape.unirest.http.exceptions.UnirestException: org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:8088 [/127.0.0.1] failed: Connection refused (Connection refused)`



`owaspzap in ~ $ curl -I -x 127.0.0.1:8088 www.google.com
HTTP/1.1 200 OK
Date: Tue, 12 Feb 2019 12:48:44 GMT
Expires: -1
Cache-Control: private, max-age=0
Content-Type: text/html; charset=ISO-8859-1
P3P: CP="This is not a P3P policy! See g.co/p3phelp for more info."
Server: gws
X-XSS-Protection: 1; mode=block
X-Frame-Options: SAMEORIGIN
Set-Cookie: 1P_JAR=2019-02-12-12; expires=Thu, 14-Mar-2019 12:48:44 GMT; path=/; domain=.google.com
Set-Cookie: NID=158=YztskfTJ-xi3xy893vunbb1LO9FKi5x5EnXBREdNF_aarGrJjoX8ePPxbt6RPm-P4ejBrd-7CA-VY_E5ikU-tQBTJ4vgPGfSYgAwpFTVziAzREDUvUglNDIa6WEELbmtrZrtqs7_-n15e43Hk-M1_-ztxNMiAFctXttG_IfJhXE; expires=Wed, 14-Aug-2019 12:48:44 GMT; path=/; domain=.google.com; HttpOnly
Accept-Ranges: none
Vary: Accept-Encoding`

please find below the pipeline I am using :

`pipeline {
    agent { label 'owaspzap' }
    stages { 
        stage('Setup') {
            steps {
                script {
                    startZap(host: "127.0.0.1", port: 8088, timeout:500, zapHome: "/usr/share/zaproxy", sessionPath:"", allowedHosts:['testserver'])
                }
            }
        }
        stage('scan') {
            steps {
                script {
                    runZapCrawler(host: "https://testserver:8080/")
                }
            }
        }
    }
    post {
        always {
            script {
                archiveZap(failAllAlerts: 1, failHighAlerts: 0, failMediumAlerts: 0, failLowAlerts: 0, falsePositivesFilePath: "zapFalsePositives.json")
            }
        }
    }
}`

I tried both on a remote agent with zap and installed zap on local agent, I get the same problem.
Thanks for any help provided

**Comment by vrondakis** [comment]
You could try running 
"/usr/share/zaproxy/zap.sh -daemon -host "127.0.0.1" -port 8088 -config api.disablekey=true -config api.addrs.addr.regex=true -config api.addrs.addr.name=10.* -config connection.timeoutInSecs=600"

directly in your console. There may be a start up error. If there isn't then try visiting http://127.0.0.1:8088 from your browser and seeing if you can access the ZAP control panel

**Comment by urand0m** [comment]
ok found out, the error lies here :

-config api.addrs.addr.name=10.* which I believe will restrict the network access to a 10.* network. mine is on 192.* for example.  is it possible to change it in the execution of the startZap?

**Comment by vrondakis** [comment]
Can you confirm that running "/usr/share/zaproxy/zap.sh -daemon -host "127.0.0.1" -port 8088 -config api.disablekey=true -config api.addrs.addr.regex=true -config api.addrs.addr.name=* -config connection.timeoutInSecs=600"

works & you are able to access the control panel?

**Comment by urand0m** [comment]
so, no with : 
`-config api.addrs.addr.name=*`

instead it does with  :
`api.addrs.addr.name=192.* `

**Comment by vrondakis** [comment]
How about -config api.addrs.addr.name=.* -config api.addrs.addr.regex=true
?

It is hard coded by mistake, I'll release an update if that works

**Comment by urand0m** [comment]
Yes I saw infact I just changed the line and recompiled.  anyway just tested with the .* and works on my side

**Comment by vrondakis** [comment]
Okay cool, I have fixed the issue in the repo and will release a new version tomorrow. Thanks for your help

**Comment by vrondakis** [comment]
Fixed in 1.9
